core/trino-main/src/main/java/io/trino/execution/scheduler/TaskExecutionClass.java

```diff
@@ -21,5 +21,13 @@ public enum TaskExecutionClass
     // Tasks from stages with some upstream stages still running.
     // To be scheduled only if no STANDARD tasks can fit.
     // Picked to kill if worker runs out of memory to prevent deadlock.
-    SPECULATIVE,
+    SPECULATIVE;
+
+    boolean canTransitionTo(TaskExecutionClass targetExecutionClass)
+    {
+        return switch (this) {
+            case STANDARD -> targetExecutionClass == STANDARD;
+            case SPECULATIVE -> targetExecutionClass == SPECULATIVE || targetExecutionClass == STANDARD;
```

Comment:
it can be `true`
if you want it to be future-proof, you can make a `switch(targetExecutionClass)` and require a code update if a new option is added

Reply:
will leave. Extra option is added later anyway